spec/services/episode_services/clean_index_spec.rb

```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+RSpec.describe EpisodeServices::CleanIndex do
+  let(:service) { EpisodeServices::CleanIndex.new.call }
+  let!(:episodes) do
+  [
+    Fabricate(:episode, indexed_at: 1.minute.ago),
+    Fabricate(:episode, indexed_at: 1.day.ago)
+  ]
+  end
+
+  it "runs indexer for all outdated the episodes" do
+    service
+
+    expect(Episode.all.map { |i| i.indexed_at }).to all(be_nil)
```

Comment:
Also, not so so good to have these sort of **tricky tricks** on tests, mostly because they can fail or give a false impression of a reliable test. But also for readability purposes.

In general, it is better to have some very imperative straightforward code here.

For this specific example is not a huge issue, but you could at least use the shorter syntax, and the episodes array.

`expect(episodes.map(&:indexed_at).to all(be_nil)`

BTW: nice touch with the `.to all(be_nill)`, never seen it before but I found it useful!

Reply:
Nice! I will use the shorter syntax, so I can use the `all(be_nil)` matcher here.